app/lib/S3Actions.scala

```diff
@@ -9,16 +9,25 @@ import com.amazonaws.services.s3.AmazonS3Client
 import com.amazonaws.services.s3.model.{ObjectMetadata, PutObjectRequest, PutObjectResult, AmazonS3Exception}
 import com.gu.pandomainauth.model.User
 
-case class S3Upload(url: URI)
+trait S3UploadResponse {
+  def fileName : String
+  def msg  : Option[String]
+}
+
+case class S3UploadSuccess(url: URI, fileName: String = null, msg: Option[String] = null)
```

Comment:
If value are nullable they should almost definitely be an option. This is a bit weird :banana: :bangbang: